docs/modules/automation-testing/pages/scripting/obfuscate-private-data-in-appium-script.adoc

```diff
@@ -0,0 +1,166 @@
+= Obfuscate private data in Appium script
+:navtitle: Obfuscate private data in Appium script
+
+Learn how to obfuscate (hide) private data in an Appium test script in the *View HTTP Headers* and *Appium Inspector* section of Session Explorer.
+
+== Data obfuscation Appium setting
+
+We introduced a new custom setting to start and stop obfuscation for private data in an Appium sesssion:
+
+[options="header"]
+|=======================
+|Appium setting | Description | Default value
+| `kobiton:privateMode` | Set to `true` to start obfuscating data, or to `false` to stop obfuscating | `false`
+|=======================
+
+== Start data obfuscation
+
+Before starting a test action that involves private data, such as passing account password, use the `Update Settings` Appium command to set `'kobiton:privateMode'` to `true`.
+
+.Example (JavaScript)
+[source,javascript]
+
+await driver.updateSettings({'kobiton:privateMode': true})
+
+As long as the setting is `true`, the data provided is marked for obfuscation.
+
+== Stop data obfuscation
+
+When test steps no longer involve private data, use the `Update Settings` Appium command to set `'kobiton:privateMode'` back to `false`.
+
+.Example (JavaScript)
+[source,javascript]
+
+await driver.updateSettings({'kobiton:privateMode': false})
+
+After this command, the data is no longer obfuscated.
+
+== Examples
+
+Below is a full example of a JavaScript code using `wd` that perform a simple login on a website and obfuscates the username and password during login:
```

Comment:
slight revision:
Below is a complete example of JavaScript code using 'wd' that demonstrates a simple login on a website, with the username and password obfuscated during the process.

Reply:
updated